Relax requirement for specification of Git provider for Repos

alexott opened the pull request:
Before this PR, the `--provider` option was required, but in many cases we can guess the provider ID from the URL (for hosted providers as Gitlab, GitHub, etc.) - this PR is making `--provider` option not required, but if we can't guess Git provider, it will error out.

victorsun123 commented on databricks_cli/sdk/service.py:
```diff
@@ -1050,10 +1051,30 @@ def stop(self, pipeline_id=None, headers=None):
+                netloc = netloc[(idx+1):]
+            provider = ReposService.__git_providers__.get(netloc.lower())
+        except:
+            pass
```
do we want to throw some error message here so it's clear to the user they need to specify the git provider if the request fails?

alexott replied:
Added it in another place instead

victorsun123 commented:
Left some comments but otherwise looks good! Also, is there any documentation for this CLI, we should make it clear in the documentation that we will parse provider if not specified and it will throw an error if unable to parse.

alexott commented:
Added some tests, added explicit exception. Doc string for `--provider` parameter was already saying that it's required if we can't detect the git provider from URL

alexott commented:
@victorsun123 any chance to review it?

nfx commented:
@alexott please rebase :)

alexott commented:
@nfx rebased

nfx commented:
@alexott tests are failing

nfx commented on tests/repos/test_cli.py:
```diff
@@ -75,6 +75,18 @@ def test_create_validation(repos_api_mock):
+    runner = CliRunner()
+    runner.invoke(cli.create_repo_cli,
+                  ["--url", TEST_URL])
+    repos_api_mock.create.assert_called_once_with(
```
can you also add a test for `awsCodeCommit`, as it has the deepest logic?..

alexott replied:
Added tests...

alexott commented:
@nfx all comments are addressed...
